### PR TITLE
Issue200 Make User Creation and Profile Creation in an atomic transaction.

### PR DESCRIPTION
**Pull request by yp1383**
Make User Creation and Profile Creation in an atomic transaction.
Corrected the logic of user creation.
Modified `test_user.py` in authentication dir.
Drop and Rebuild the database on Heroku.
Now even though `profile_id` and `user_id` do not match, it can still successfully load the profile page.
ex. `user_id` = 10, `profile_id` = 9. -> `profliepage/10/` will load successfully.

### CODE REVIEW

File: dayplanner/settings.py [hunk 1]
```diff
@@ -33,10 +33,7 @@
 
 
 # SECURITY WARNING: don't run with debug turned on in production!
-if os.environ.get("DEBUG", "true") == "true":
-    DEBUG = True
-else:
-    DEBUG = False
+DEBUG = True
```
Comment (sballan) [inline dayplanner/settings.py:36]: Don't remove this, we want the DEBUG changes

File: dayplanner/settings.py [hunk 2]
```diff
@@ -33,10 +33,7 @@
 
 
 # SECURITY WARNING: don't run with debug turned on in production!
-if os.environ.get("DEBUG", "true") == "true":
```
Comment (sballan) [inline dayplanner/settings.py:36]: Lines 36-39 should not be changed

File: dayplanner/settings.py [hunk 3]
```diff
@@ -67,8 +64,8 @@
 ]
 
 MIDDLEWARE = [
-    "django.middleware.security.SecurityMiddleware",
     "whitenoise.middleware.WhiteNoiseMiddleware",
```
Comment (sballan) [inline dayplanner/settings.py:67]: Don't change this

File: dayplanner/settings.py [hunk 4]
```diff
@@ -171,10 +168,18 @@
 STATIC_URL = "/static/"
 STATIC_ROOT = os.path.join(BASE_DIR, "staticfiles")
 STATICFILES_DIRS = (os.path.join(BASE_DIR, "dayplanner/static"),)
-STATICFILES_STORAGE = "whitenoise.storage.CompressedManifestStaticFilesStorage"
```
Comment (sballan) [inline dayplanner/settings.py:174]: Don't change this

**Comment by yp1383** [comment]
Ok. I believe when I merged the dev branch, I forget to git pull first. so somehow the setting is different from the dev branch.
I fixed the error now.

**Comment by wangsnuxiao** [comment]
Great! Happy to see everything is fixed

**Review by wangsnuxiao — APPROVED** [review]

**Pull request merged**